pandas/core/reshape/concat.py

```diff
@@ -262,7 +262,13 @@ def __init__(self, objs, axis=0, join='outer', join_axes=None,
         ndims = set()
         for obj in objs:
             if not isinstance(obj, NDFrame):
-                raise TypeError("cannot concatenate a non-NDFrame object")
+
+                # multiline string concat is safer with whitespace for regex
+                # at test_concat.py:181
+                msg = """cannot concatenate object of type "{0}";"""
+                msg += " only pd.Series, pd.DataFrame, and pd.Panel"
+                msg += " objs are valid".format(type(obj))
```

Comment:
Can you write this as 

```
msg = (" ..."
       " ... ")
```

(which is the typical way we write multi-line strings in the codebase)